doc/source/examples/unresolved-cfd-dem/dense-pneumatic-conveying/dense-pneumatic-conveying.rst

```diff
@@ -0,0 +1,636 @@
+
+    subsection boundary conditions
+      set number = 2
+      subsection bc 0
+        set id   = 0
+        set type = function weak
+        set beta = 100
+        subsection u
+          set Function expression = 0
+        end
+        subsection v
+          set Function expression = 0
+        end
+        subsection w
+          set Function expression = 0
+        end
+      end
+      subsection bc 1
+        set id                 = 1
+        set type               = periodic
+        set periodic_id        = 2
+        set periodic_direction = 0
+      end
+    end
+
+Flow control
+~~~~~~~~~~~~~~~~~~
+
+Since the simulation has periodic boundary conditions, a correction force is needed to drive the flow to compensate the pressure drop in the pipe. In other to achieve this, we use the `dynamic flow controller <../../../parameters/cfd/dynamic_flow_control.html>`_. Here, we also apply a proportional beta force on particles. The average velocity is set to 3 m/s, this correspond to the average over the entire domain considering the void fraction. The flow controller performs well for CFD simulation, but needs some tuning for CFD-DEM simulation. By default, the controller has a high stiffness and aims to correct the flow in the next time step. However, the carrying of particles by the flow leads to a response time that is not taken into account and results in a oscillation of the velocity of the flow. To avoid this, we use the beta threshold and the alpha relaxation parameter. Here, the beta value will not be updated if the new beta value is within the 5% of the previous value. Also, the correction to apply to the previous beta value is reduced by a factor of 0.25. This way, the velocity of the flow and the particles are more stable.
+
+.. code-block:: text
+
+   subsection flow control
+     set enable               = true
+     set enable beta particle = true
+     set average velocity     = 3
+     set flow direction       = 0
+     set beta threshold       = 0.05
+     set alpha                = 0.25
+     set verbosity            = verbose
+   end
+
+Void Fraction
+~~~~~~~~~~~~~~~
+
+We choose the `quadrature centred method (QCM) <../../../theory/multiphase/cfd_dem/unresolved_cfd-dem.html#the-quadrature-centered-method>`_  to calculate the void fraction. The ``l2 smoothing factor`` we choose is the square of twice the diameter of the particles.
+
+.. code-block:: text
+
+    subsection void fraction
+      set mode                = qcm
+      set read dem            = true
+      set dem file name       = dem
+      set l2 smoothing factor = 0.0001
+    end
+
+CFD-DEM
+~~~~~~~~~~
+
+The chosen drag model is Di Felice, and we use the Saffman lift force, the buoyancy force, and the pressure force. The coupling frequency is set to 100, which means that the DEM time step is 5e-6 s, for a Rayleigh critical time step of about 3.5 %. The grad-div stabilization is used with a length scale of 0.084, the diameter of the pipe.
```

Comment:
```suggestion
We use the Di Felice drag model, the Saffman lift force, the buoyancy force, and the pressure force. The coupling frequency is set to 100, which means that the DEM time step is 5e-6 s. The DEM time-step is 3.5% of the Rayleigh critical time step. The grad-div stabilization is used with a length scale of 0.084, the diameter of the pipe.
```

Reply:
Fun fact, but I looked what is the right way to write "time-step" since I also though it was with a hyphen and the right ways are timestep or time step